client/src/app/appCtrl.js

```diff
@@ -546,4 +555,62 @@ angular.module('app').controller('appCtrl', ['$scope', '$http', '$timeout', 'app
     });
   });
 
+
+  // given values in highlighting{} structure, 
+  // highlight areas where a select function value crosses a threshold
+  // used in dygraph's underlayCallback
+  function highlightAnomaly(canvas, area, g) {
+    // sync semaphor, do not start another call till last finished
+    if($scope.view.highlighting.finished === false) {
+      return;
+    }
+    console.log("highlighting...");
+    $scope.view.highlighting.finished = false;
+    var timeIdx;
+
+    // draw rectangle on x0..x1
+    function highlight_period(x_start, x_end, color) {
+      var canvas_left_x = g.toDomXCoord(x_start);
+      var canvas_right_x = g.toDomXCoord(x_end);
+      var canvas_width = canvas_right_x - canvas_left_x;
+      canvas.fillStyle = color;
+      canvas.fillRect(canvas_left_x, area.y, canvas_width, area.h);
+    }
+
+    // find x values matching condition on y-value
+    // params: data (all fields), watchedFieldName (string), threshold (for condition >thr)
+    // return array with indices of anomalies
+    function find_where(data, watchedFieldName, threshold, dt) {
+      var results = [];
+      var fnIdx = loadedFields.indexOf(watchedFieldName);
+      if (fnIdx === -1) {
+        handleError("Highlighting cannot work, field "+watchedFieldName+" not found!", "danger", true);
+        return results;
+      }
+      timeIdx = loadedFields.indexOf(appConfig.TIMESTAMP);
+      for( var i=0; i< data.length; i++) {
+        var row = data[i];
+        // the condition is here
+        if (row[fnIdx] >= threshold) {
+          var time = row[timeIdx];
+          //console.log("found anomaly at "+time+" with value "+row[fnIdx]);
+          results.push(time);
+        }
+      }
+      return results;
+    } //end find_where
+
+    //highlight_period(2, 5, yellow); //test
+    // find relevant points
+    var selected = find_where(loadedCSV, $scope.view.highlighting.fieldName, $scope.view.highlighting.threshold);
+    var modDt = $scope.view.highlighting.radius*(loadedCSV[1][timeIdx]-loadedCSV[0][timeIdx]); // update dt to grain resol of timestamps
```

Comment:
actually, this works OK, as the `timestamp` is always present (if not, we add it artificially)